데이터 패칭 시 오류 해결 방법

axios 라이브러리는 오류를 만들어서 catch 문으로 보내주지만 
fetch는 오류를 따로 잡아주지 못해서 response의 ok 혹은 status를 사용해서 수동으로 오류로 보내야한다.
오류를 발생시켰을 경우 즉시 다음 실행을 중단하고 catch로 넘어가게된다.

1. then을 사용할 경우에는 catch 함수를 사용해서 오류를 잡는다.

ex) 

```javascript
const dataHandler = () => {
    fetch(url)
        .then(response => {
            if(!response.ok) {
                throw new Error('오류 발생')
            }
            return response.json();
        })
        .then(data => {
            // 데이터에 대한 처리
        })
        .catch(error => {
            // 오류에 대한 처리
        })
}
```

2. async/await를 사용할 경우 try/catch문을 사용해서 오류를 잡는다.

ex)
```javascript
const dataHandler = async() => {
    try {
        const response = await fetch(url);

        if(!response.ok) {
            throw new Error('오류 발생')
        }

        const data = await response.json();

        // 데이터에 대한 처리
    } catch(error) {
        // 오류에 대한 처리
    }
}
```